Massive changeling overhaul holy shit

Crystalwarrior opened the pull request:
Thanks to @Xhuis for the original PR ( https://github.com/tgstation/-tg-station/pull/13138 )!

![Current tier costs](http://puu.sh/lCS2q/131889fdd0.png)
^Above image shows all cost changes and absorbtion requirements to the changeling abilities
- experimental: "Complete changeling rework! Now changeling abilities are divided into tiers based on required amount of absorbtions to unlock them. You start with 5 Evolution Points by default and have access to Tier 1, which includes various stings. You also have lesser form and mimic voice abilities from the start."
- experimental: "This is a VERY experimental update, and all player feedback is much appreciated!"
- rscremove: "Removed Changeling Shield, Chitinous Armor and Extract DNA sting"
- rscremove: "Removed Changeling team objectives"
- rscadd: "Added synaptic enhancement as a changeling ability. When active, this will halve the delay between actions such as attacking."
- rscadd: "Paralysis Sting has been re-added and revamped. After a short delay based on the victim's health, they will be knocked down for fifteen seconds."
- rscadd: "Death Sting has been re-added and revamped. It will inject them with a small dosage of Wasting Toxin with slow metabolization rate, which will slowly drain blood and deal toxin damage to the victim."
- rscadd: "Horror Form has been added. When used, this will transform the changeling into their powerful true form, able to deal powerful damage, crawl through vents, and tear into living and dead humans to heal themselves. They will be incapable of returning to a human for three minutes, and when they do, are stunned for ten seconds. Death will automatically turn them back into humans and knock them out for thirty seconds."
- tweak: "Chameleon ability for Changelings now makes them invisible to mouse cursor when they're completely invisible."
- tweak: "Chameleon ability has been sped up slightly."
- tweak: "Evolution points have received a large rebalance. Changelings now start with five points, and will gain two more after each absorption. Some abilities take more than five points to unlock, meaning powerful abilities like Fleshmend will be locked by default. If a changeling absorbs another changeling, the absorber will gain the other's remaining genetic points."
- tweak: "There is now no limit on genomes storable for transformation. This means that you will be able to transform into anyone you have absorbed past the first six."
- tweak: "Mute Sting now functions based on the Mute Toxin reagent and lasts for much shorter."
- tweak: "Blind Sting now confuses the victim for a short duration."
- tweak: "All stings but transformation sting no longer function on the dead. Cryogenic Sting now only works on conscious targets."
- bugfix: "Fixed transformations sometimes giving you 'fake' items that prevented you from equipping the real deal"
- bugfix: "Fixed lesser form dropping ling eye implants"

Crystalwarrior committed:
Initial commit

Crystalwarrior committed:
holy shit.
Chameleon mutation for Changelings now makes their mouse_opacity 0 when they're completely invisible.
Chameleon mutation sped up
All changeling abilities are now tier-based
Fuck, just wait for me to update the PR.

TDcoolguy300 commented on code/game/gamemodes/changeling/changeling.dm:
```diff
@@ -421,12 +423,12 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 	user.underwear = chosen_prof.underwear
 	user.undershirt = chosen_prof.undershirt
 	user.socks = chosen_prof.socks
-	user.dna = chosen_dna //Should fix the whole "ERM MEH GERD I TURNED INTO A HUMAN INSTEAD OF ROBIT"
```
We've got an issue here Crystal, that was there to deal with changlings not getting the correct species, unless you managed to handle it elsewhere now.

TDcoolguy300 commented on code/game/gamemodes/changeling/changeling.dm:
```diff
@@ -421,12 +423,12 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 	user.updateappearance(mutcolor_update=1)
 	user.update_body()
 	user.domutcheck()
-/* //Seriously, why was this even added?
```
Wasnt this suppose to be commented out?

Crystalwarrior replied:
I re-commented it out again, hue

Crystalwarrior committed:
Removed weird "fake flesh items" given to you when you transform into someone.
Removed swap forms ability due to it's uselessness
Tweaked synaptic enhancement to cost 30 chems + halting regeneration rate completely, upped the cost of it
Fixed lesser form dropping ling cyberimplants
Biodegrade moved to Tier 3, cost upped
misc fixes etc.

Crystalwarrior committed:
fixed an issue pointed out by tdcoolguy

Crystalwarrior committed:
WEW

Crystalwarrior committed:
Changelog

Crystalwarrior committed:
Updated changelog

Crystalwarrior committed:
Death sting fun time

Crystalwarrior committed:
aaa fuck off JS errors

Crystalwarrior committed:
fixed positioning of UI items

Crystalwarrior committed:
New sting sprites

Crystalwarrior committed:
credits for the slaves :^)))))))))))))))
